ffi/src/expressions.rs

```diff
@@ -146,7 +146,8 @@ fn visit_expression_column_impl(
     state: &mut KernelExpressionVisitorState,
     name: DeltaResult<String>,
 ) -> DeltaResult<usize> {
-    Ok(wrap_expression(state, Expression::Column(name?)))
+    // TODO: We can't actually assume the column name is so easily splittable!
+    Ok(wrap_expression(state, Expression::split_column(name?)))
```

Comment:
Out of curiosity, what are cases when a column name isn't splittable?

Reply:
See other comment... we can't safely split an arbitrary column name like `"a.b.c.d"` because one of the field names might be `b.c` (probably not, but no way to rule it out).